File: trains-gdx-core/src/main/java/com/github/caay2000/trains/world/WorldSize.java
package com.github.caay2000.trains.world;

import java.util.Set;

public class WorldSize {

    private int maxX = 0;
    private int maxY = 0;
    private int minX = 0;
    private int minY = 0;

    public WorldSize(Set<? extends WorldObject> objects) {

        for (WorldObject obj : objects) {
            updateSize(obj);
        }
    }

    public int getMinX() {
        return minX;
    }

    public int getMinY() {
        return minY;
    }

    private void updateSize(WorldObject obj) {
        if (obj.getPosition().getX() > maxX) {
            maxX = (int) obj.getPosition().getX();
        }
        if (obj.getPosition().getX() < minX) {
            minX = (int) obj.getPosition().getX();
        }
        if (obj.getPosition().getY() > maxY) {
            maxY = (int) obj.getPosition().getY();
        }
        if (obj.getPosition().getY() < minY) {
            minY = (int) obj.getPosition().getY();
        }
    }

}

File: trains-gdx-core/src/main/kotlin/com/github/caay2000/trains/world/WorldSize.kt
package com.github.caay2000.trains.world

import com.github.caay2000.trains.world.aaa.WorldEntity

class WorldSize(objects: Set<WorldEntity>) {
    private var maxX = 0
    private var maxY = 0
    var minX = 0
        private set
    var minY = 0
        private set

    private fun updateSize(obj: WorldEntity) {
        if (obj.position().x > maxX) {
            maxX = obj.position().x.toInt()
        }
        if (obj.position().x < minX) {
            minX = obj.position().x.toInt()
        }
        if (obj.position().y > maxY) {
            maxY = obj.position().y.toInt()
        }
        if (obj.position().y < minY) {
            minY = obj.position().y.toInt()
        }
    }

    init {
        for (obj in objects) {
            updateSize(obj)
        }
    }
}